coordinator/internal/meshapi/meshapi_test.go

```diff
@@ -0,0 +1,134 @@
+	"encoding/pem"
+	"log/slog"
+	"testing"
+
+	"github.com/edgelesssys/contrast/coordinator/internal/authority"
+	"github.com/edgelesssys/contrast/coordinator/internal/seedengine"
+	"github.com/edgelesssys/contrast/internal/ca"
+	"github.com/edgelesssys/contrast/internal/manifest"
+	"github.com/edgelesssys/contrast/internal/platforms"
+	"github.com/edgelesssys/contrast/internal/testkeys"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+)
+
+func TestNewMeshCert(t *testing.T) {
+	m := manifestWithWorkloadOwnerKey(t, testkeys.ECDSA(t))
+	policyHash := sha256.Sum256(nil)
+	policyHashHex := manifest.NewHexString(policyHash[:])
+	m.Policies = map[manifest.HexString]manifest.PolicyEntry{
+		policyHashHex: {
+			SANs:             []string{"test"},
+			WorkloadSecretID: "test",
+		},
+	}
+	key := testkeys.New[ecdsa.PrivateKey](t, testkeys.ECDSAP384Keys[0])
+	meshKey := testkeys.New[ecdsa.PrivateKey](t, testkeys.ECDSAP384Keys[1])
+	rootKey := testkeys.New[ecdsa.PrivateKey](t, testkeys.ECDSAP384Keys[2])
+
+	seed := [32]byte{}
+	salt := [32]byte{}
+	se, err := seedengine.New(seed[:], salt[:])
+	require.NoError(t, err)
+	ca, err := ca.New(rootKey, meshKey)
+	require.NoError(t, err)
+
+	info := authority.AuthInfo{
+		TLSInfo: credentials.TLSInfo{
+			State: tls.ConnectionState{
+				PeerCertificates: []*x509.Certificate{{PublicKey: key.Public(), PublicKeyAlgorithm: x509.ECDSA}},
+			},
+		},
+		Report: &fakeReport{
+			hostData: policyHash[:],
+		},
+		State: &authority.State{
+			Manifest:   m,
+			SeedEngine: se,
+			CA:         ca,
+		},
+	}
+	ctx := peer.NewContext(context.Background(), &peer.Peer{
+		AuthInfo: info,
+	})
+
+	meshapi := New(slog.Default())
+
+	resp, err := meshapi.NewMeshCert(ctx, nil)
+	require.NoError(t, err)
+
+	require.NotEmpty(t, resp.WorkloadSecret)
+
+	certChain := certFromPEM(t, resp.CertChain)
+	cert, intermediateCert := certChain[0], certChain[1]
+	assert.False(t, cert.IsCA)
+	assert.True(t, intermediateCert.IsCA)
+	assert.Equal(t, cert.AuthorityKeyId, intermediateCert.SubjectKeyId)
+
+	meshCert := certFromPEM(t, resp.MeshCACert)[0]
+	assert.True(t, meshCert.IsCA)
+	assert.Equal(t, cert.AuthorityKeyId, meshCert.SubjectKeyId)
+	assert.Empty(t, meshCert.AuthorityKeyId)
+
+	rootCert := certFromPEM(t, resp.RootCACert)[0]
+	assert.True(t, rootCert.IsCA)
+	assert.Empty(t, rootCert.AuthorityKeyId)
+	assert.Equal(t, intermediateCert.AuthorityKeyId, rootCert.SubjectKeyId)
+}
+
+type fakeReport struct {
+	extensions []pkix.Extension
+	hostData   []byte
+}
+
+func (r *fakeReport) ClaimsToCertExtension() ([]pkix.Extension, error) {
+	return r.extensions, nil
+}
+
+func (r *fakeReport) HostData() []byte {
+	return r.hostData
+}
+
+func manifestWithWorkloadOwnerKey(t *testing.T, key *ecdsa.PrivateKey) *manifest.Manifest {
+	t.Helper()
+	require := require.New(t)
+
+	m, err := manifest.Default(platforms.AKSCloudHypervisorSNP)
```

Comment:
Do we even need a default manifest here, especially since we use a constant platform here? Do we even need the workload owner key in the manifest? As far as I can see `NewMeshCert()` only uses `state.Manifest.Policies[hostData]`

Reply:
I copied this over from `userapi_test.go` without thinking too much. You're right that the workload owner key is not used so I'm inlining the `manifest.Default` as suggested.

Reply:
My idea was that we might even not call into `Default`, also because this means that the tests are not executable with plain Go:
```
euler@work:~/projects/coco/coordinator/internal/meshapi$ go test ./...
--- FAIL: TestNewMeshCert (0.00s)
panic: failed to unmarshal embedded reference values: json: cannot unmarshal string into Go value of type manifest.EmbeddedReferenceValues [recovered]
        panic: failed to unmarshal embedded reference values: json: cannot unmarshal string into Go value of type manifest.EmbeddedReferenceValues
```

Reply:
Oh, that's even nicer - thanks; done!